src/screens/Profile/PaymentOptions/index.tsx

```diff
@@ -0,0 +1,205 @@
+import SvgDelete from '@atb/assets/svg/mono-icons/actions/Delete';
+import {useAuthState} from '@atb/auth';
+import FullScreenHeader from '@atb/components/screen-header/full-header';
+import ThemeText from '@atb/components/text';
+import PaymentBrand from '@atb/screens/Ticketing/Purchase/Payment/PaymentBrand';
+import {
+  getExpireDate,
+  getPaymentTypeName,
+} from '@atb/screens/Ticketing/Purchase/utils';
+import {StyleSheet, Theme, useTheme} from '@atb/theme';
+import {
+  deleteRecurringPayment,
+  listRecurringPayments,
+  RecurringPayment,
+} from '@atb/tickets';
+import {useTranslation} from '@atb/translations';
+import PaymentOptionsTexts from '@atb/translations/screens/subscreens/PaymentOptions';
+import React, {useEffect, useState} from 'react';
+import {ActivityIndicator, TouchableOpacity, View} from 'react-native';
+import {ScrollView} from 'react-native-gesture-handler';
+import {destructiveAlert} from '../Home/utils';
+import MessageBox from '@atb/components/message-box';
+import {StackNavigationProp} from '@react-navigation/stack';
+import {RootStackParamList} from '@atb/navigation';
+import useFontScale from '@atb/utils/use-font-scale';
+
+type PaymentOptionsProps = {
+  navigation: StackNavigationProp<RootStackParamList>;
+};
+
+export default function PaymentOptions({navigation}: PaymentOptionsProps) {
+  const style = useStyle();
+  const {theme} = useTheme();
+  const {t} = useTranslation();
+  const {user} = useAuthState();
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [storedCards, setStoredCards] = useState<RecurringPayment[]>([]);
+
+  async function getRecurringPaymentOptions(): Promise<
+    Array<RecurringPayment>
+  > {
+    if (!user || user.isAnonymous) return [];
+    const recurringOptions: Array<RecurringPayment> =
+      await listRecurringPayments();
+    return [...recurringOptions.reverse()];
```

Comment:
The `[...obj]` copying here is just extra work for copying where it has no real effect. listRecurringPaymets returns newly created data and isn't shared anywhere. Also `.reverse()` mutates the array itself before copying it. If you really wanted to reverse a copy it would have to be `[...recurringOptions].reverse()`, but still I don't think this is needed.

Reply:
I agree. This was a copy paste from elsewhere in the app. I assumed it did this for a reason. 😄